module-bsp/board/rt1051/puretx/hal/battery_charger/BatteryCharger.cpp

```diff
@@ -73,7 +81,7 @@ namespace hal::battery
                 bsp::battery_charger::clearFuelGuageIRQ(
                     static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::minVAlert));
 
-                eventsHandler.onBrownout();
+                eventsHandler.onBrownout(bsp::battery_charger::getChargeStatus());
```

Comment:
Wouldn't be better to include all the implemented logic here? (BatteryCharger.cpp for pure). Brownout detector and interface of BatteryCharger does not have to know about existence of fuel gauge IRQs - this is more low level concept. Also Bell have different logic for different chips.

Reply:
done